src/Infusionsoft/Infusionsoft.php

```diff
@@ -433,27 +475,40 @@ public function request()
         // Before making the request, we can make sure that the token is still
         // valid by doing a check on the end of life.
         $token = $this->getToken();
-        if ($this->isTokenExpired()) {
+        if ($this->authenticationType === AuthenticationType::OAuth2AccessToken && $this->isTokenExpired()) {
             throw new TokenExpiredException;
         }
 
-        $url = $this->url . '?' . http_build_query(array('access_token' => $token->getAccessToken()));
-
         $params = func_get_args();
         $method = array_shift($params);
 
         // Some older methods in the API require a key parameter to be sent
         // even if OAuth is being used. This flag can be made false as it
         // will break some newer endpoints.
         if ($this->needsEmptyKey) {
-            $params = array_merge(array('key' => $token->getAccessToken()), $params);
+            $params = array_merge(array('key' => 'backwards-compatability'), $params);
         }
 
         // Reset the empty key flag back to the default for the next request
         $this->needsEmptyKey = true;
 
+        $options = [];
+
+        if ( $this->authenticationType === AuthenticationType::OAuth2AccessToken ) {
```

Comment:
Since the same code is being used below (line 542-550), should this be a function?